percona/controller/pgcluster/backup.go

```diff
@@ -0,0 +1,151 @@
+		if err != nil {
+			return errors.Wrap(err, "failed to list backup jobs")
+		}
+
+		for _, job := range backupJobs.Items {
+			if err := reconcileBackupJob(ctx, r.Client, cr, job, repo.Name); err != nil {
+				return errors.Wrap(err, "failed to reconcile backup job")
+			}
+		}
+	}
+	return nil
+}
+
+func reconcileBackupJob(ctx context.Context, cl client.Client, cr *v2.PerconaPGCluster, job batchv1.Job, repoName string) error {
+	pb, err := findPGBackup(ctx, cl, cr, job)
+	if err != nil {
+		return errors.Wrapf(err, "failed to find PerconaPGBackup for job %s", job.Name)
+	}
+
+	if pb == nil {
+		canCreate, err := canCreatePGBackup(ctx, cl, cr)
+		if err != nil {
+			return errors.Wrap(err, "failed to check if we can create PerconaPGBackup")
+		}
+		if !canCreate {
+			return nil
+		}
+
+		// Create PerconaPGBackup resource for backup job,
+		// which was created without this resource.
+		pb = &v2.PerconaPGBackup{
+			ObjectMeta: metav1.ObjectMeta{
+				GenerateName: job.Name + "-",
+				Namespace:    job.Namespace,
+				Annotations: map[string]string{
+					v2.AnnotationPGBackrestBackupJobName: job.Name,
+				},
+			},
+			Spec: v2.PerconaPGBackupSpec{
+				PGCluster: cr.Name,
+				RepoName:  repoName,
+			},
+		}
+
+		err = cl.Create(ctx, pb)
+		if err != nil {
+			return errors.Wrapf(err, "failed to create PerconaPGBackup for job %s", job.Name)
+		}
+	}
+
+	err = retry.RetryOnConflict(retry.DefaultRetry, func() error {
+		j := new(batchv1.Job)
+		err := cl.Get(ctx, types.NamespacedName{Name: job.Name, Namespace: job.Namespace}, j)
+		if err != nil {
+			return err
+		}
+		j.OwnerReferences = nil
+		if err := controllerutil.SetControllerReference(pb, j, cl.Scheme()); err != nil {
+			return errors.Wrap(err, "failed to set controller reference")
+		}
+
+		return cl.Update(ctx, j)
+	})
+	if err != nil {
+		return errors.Wrap(err, "failed to update backup job")
+	}
+
+	return nil
+}
+
+func findPGBackup(ctx context.Context, cl client.Reader, cr *v2.PerconaPGCluster, job batchv1.Job) (*v2.PerconaPGBackup, error) {
+	pbList, err := listPGBackups(ctx, cl, cr)
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to list backup jobs")
+	}
+
+	for _, pb := range pbList.Items {
+		pb := pb
+		if pb.GetAnnotations()[v2.AnnotationPGBackrestBackupJobName] == job.Name || pb.Status.JobName == job.Name {
+			return &pb, nil
+		}
+	}
+	return nil, nil
+}
+
+func canCreatePGBackup(ctx context.Context, cl client.Reader, cr *v2.PerconaPGCluster) (bool, error) {
+	pbList, err := listPGBackups(ctx, cl, cr)
+	if err != nil {
+		return false, errors.Wrap(err, "failed to list backup jobs")
+	}
+	for _, pb := range pbList.Items {
+		// We can't create resource if there is at least one in new or starting state.
+		// These states are the only ones in which the resource doesn't have `.status.jobName`.
+		if pb.Status.State == v2.BackupNew || pb.Status.State == v2.BackupStarting {
+			return false, nil
+		}
+	}
+	return true, nil
+}
```

Comment:
is this a check to prevent parallel backups? if yes, shouldn't we also check running jobs?

Reply:
It wasn't for parallel backups, and actually we don't need this function: https://github.com/percona/percona-postgresql-operator/pull/524/commits/e5321d6922dee63543ccc827cdc1488590843ce1